packages/aws-cdk/lib/api/deployments/hotswap-deployments.ts

```diff
@@ -414,36 +427,56 @@ function makeRenameDifference(
  * Returns a `NonHotswappableChange` if the change is not hotswappable
  */
 function isCandidateForHotswapping(
-  change: cfn_diff.ResourceDifference,
   logicalId: string,
+  change: cfn_diff.ResourceDifference,
+  evaluateCfnTemplate: EvaluateCloudFormationTemplate,
 ): RejectedChange | ResourceChange {
   // a resource has been removed OR a resource has been added; we can't short-circuit that change
   if (!change.oldValue) {
     return {
       hotswappable: false,
-      resourceType: change.newValue!.Type,
-      logicalId,
-      reason: NonHotswappableReason.RESOURCE_CREATION,
-      description: `resource '${logicalId}' was created by this deployment`,
+      change: {
+        reason: NonHotswappableReason.RESOURCE_CREATION,
+        description: `resource '${logicalId}' was created by this deployment`,
+        subject: {
+          type: 'Resource',
+          logicalId,
+          resourceType: change.newValue!.Type,
+          metadata: evaluateCfnTemplate.metadataFor(logicalId),
+        },
+      },
     };
   } else if (!change.newValue) {
     return {
       hotswappable: false,
-      resourceType: change.oldValue!.Type,
       logicalId,
-      reason: NonHotswappableReason.RESOURCE_DELETION,
-      description: `resource '${logicalId}' was destroyed by this deployment`,
+      change: {
+        reason: NonHotswappableReason.RESOURCE_DELETION,
+        description: `resource '${logicalId}' was destroyed by this deployment`,
+        subject: {
+          type: 'Resource',
+          logicalId,
+          resourceType: change.oldValue?.Type,
```

Comment:
nit (maybe): this used to be:

```ts
change.oldValue!.Type
```

So now we allow null values where we used to fail on and proceeding with this code path could be unexpected. I recognize this is probably just type-checking blindspots but wondering if there's something better to do here.